### PR TITLE
Bs2 63

### PR DESCRIPTION
**Pull request by pmaganty**
Create TypeORM interface and backend structure

### CODE REVIEW

**Review by SummerTinio — APPROVED** [review]

File: src/backend/controller/user/user.ts [hunk 1]
```diff
@@ -1,11 +1,38 @@
...
-  user
-) => {
-  res.status(200).json({ name: user.attributes.given_name });
+const addNewUser = async (req: NextApiRequest, res: NextApiResponse) => {
+  console.log(process.env.CONNECTION_STRING);
+  const db = await connection();
+  const newUser = await db.createQueryBuilder()
+          .insert()
+          .into(User)
+          .values([
```
Comment (bennettl) [inline src/backend/controller/user/user.ts:12]: Use request body

File: src/backend/model/entities/Interest.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+@Entity()
+export class Interest {
+
+    @PrimaryGeneratedColumn()
+    id: number;
+
+    @Column("text")
+    interest: string;
+
+    @ManyToOne(() => User, user => user.interests)
```
Comment (bennettl) [inline src/backend/model/entities/Interest.ts:13]: Should we have a separate user interest table?

File: src/pages/_middleware.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import type { NextFetchEvent, NextRequest } from 'next/server'
+import { createConnection, getConnection } from "typeorm";
+
+
+
+export function middleware(req: NextRequest, ev: NextFetchEvent) {
+
+  /*const connect = async () => {
```
Comment (bennettl) [inline src/pages/_middleware.ts:8]: Remove code that's not being used

File: src/pages/api/projects/project/[id].ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+type Data = {
+  name: string;
+};
+
+export default function handler(
+    req: NextApiRequest,
+    res: NextApiResponse<Data>
+) {
+    //Get single project
+    if (req.method === 'GET') {
```
Comment (bennettl) [inline src/pages/api/projects/project/[id].ts:13]: Seems like switch statement will be cleaner

File: src/backend/model/entities/Interest.ts (hunk 2 above)
Comment (pmaganty) [reply src/backend/model/entities/Interest.ts:13]: Actually I ended up adding the Interests table back to the typeorm connection initialize call - it breaks our DB setup if we don't. I can go ahead and re-design the DB to work without the interest table moving forward - let me know if you want that changed now

**Pull request merged**